File: bench/promptgreen/tasks/t25/py/hidden_tests.py
import os
import subprocess
import sys

HERE = os.path.dirname(os.path.abspath(__file__))
REF = os.path.join(HERE, "reference.py")

out = subprocess.run(
    [sys.executable, REF], capture_output=True, text=True, check=True
).stdout
lines = out.splitlines()

assert lines[0] == "2" and lines[1] == "3", "9875 -> root 2, persistence 3"
assert lines[2] == "7" and lines[3] == "0", "single digit input, no steps"
assert lines[4] == "0" and lines[5] == "0", "edge case: zero input"

print("all checks passed")

File: bench/promptgreen/tasks/t25/hidden_tests.mjs
// Hidden test for t25 (digital root and additive persistence). Never shown to an author.
const EXPECTED = '2\n3\n7\n0\n0\n0\n';

export async function run(compileFn, source) {
  const r = compileFn(source);
  if (r.ok !== true) {
    return { green: false, detail: { expected: EXPECTED, got: r.stdout, ok: r.ok, reason: 'compile/run failed' } };
  }
  const lines = r.stdout.split('\n');
  const check1 = lines[0] === '2' && lines[1] === '3'; // 9875 -> root 2, persistence 3
  const check2 = lines[2] === '7' && lines[3] === '0'; // single digit input, no steps
  const check3 = lines[4] === '0' && lines[5] === '0'; // edge case: zero input
  const green = check1 && check2 && check3 && r.stdout === EXPECTED;
  return { green, detail: { expected: EXPECTED, got: r.stdout, ok: r.ok, check1, check2, check3 } };
}
